### ISSUE TITLE
Add support for Synapses to Synapses

### ISSUE THREAD

**Issue by denisalevi** [brian-team/brian2cuda#133]
In brian2 one can create `Synapses` object which target `Synapses` objects. We currently get an 
```
NotImplementedError: Cannot retrieve the values of state variables in standalone code before the simulation has been run.
```
 error. This fails a bunch of test suite tests.
The reason is, that we need the `IDs` of the target group of a SynapticPathway in order to create our connectivity matrix (which is split by groups of post IDs). To get those, we need access to state variables of the `SynapticPathway` objects `target` group, which is a `Synapse` in the case of synapses to synapses connections. And we can't access `Synapse` variables before synapse creation and our current implementation fills those target group variables from python side (before the network is run), which gives us aboves error. I've tried to fix this but it was not straight forward and I'm moving on to more urgent tasks. Check the `synapses_to_synapses_support` branch for my first try. It needs some rearranging, such that the `SynapticPathway` classes (in cpp) `target_start/target_end` variables are some variables that are set from cpp side after synapse generation templates are run.

Maybe we also don't need my additional parameters in the `SynapticPathway` class (see commits below) and can just work with the `owner.target.start` variable in the `synapses_initialise_template`.

**Comment by denisalevi** [comment]
Maybe relevant? brian-team/brian2@6e8acd1b

**Comment by denisalevi** [comment]
Here are two PRs I came across related to Brian2 changes for synapse to synapse connections: brian-team/brian2#715 and brian-team/brian2#854.

**Comment by denisalevi** [comment]
This was fixed in PR #186.

Closing.
